src/pe_source/data/pe_db/config.py

```diff
@@ -0,0 +1,78 @@
+"""Get PE Source API credentials."""
+
+# Standard Python Libraries
+from configparser import ConfigParser
+import logging
+import os
+
+# Third-Party Libraries
+from importlib_resources import files
+import requests
+import shodan
+
+# Configuration
+REPORT_DB_CONFIG = files("pe_reports").joinpath("data/database.ini")
+
+
+def shodan_api_init():
+    """Connect to Shodan API."""
+    section = "shodan"
+    api_list = []
+    if os.path.isfile(REPORT_DB_CONFIG):
+        parser = ConfigParser()
+        parser.read(REPORT_DB_CONFIG, encoding="utf-8")
+        if parser.has_section(section):
+            params = parser.items(section)
+        else:
+            raise Exception(
+                "Section {} not found in the {} file".format(section, REPORT_DB_CONFIG)
+            )
+    else:
+        raise Exception(
+            "Database.ini file not found at this path: {}".format(REPORT_DB_CONFIG)
+        )
+
+    for key in params:
+        try:
+            api = shodan.Shodan(key[1])
+            # Test api key
+            api.info()
+        except Exception:
+            logging.error("Invalid API key: {}".format(key))
+            continue
+        api_list.append(api)
+    logging.info("Number of valid API keys: {}".format(len(api_list)))
```

Comment:
I always err on the side of being more specific when logging:
```suggestion
            logging.error("Invalid Shodan API key: {}".format(key))
            continue
        api_list.append(api)
    logging.info("Number of valid Shodan API keys: {}".format(len(api_list)))
```